spec/ransack/search_spec.rb

```diff
@@ -271,12 +271,21 @@ module Ransack
           parent_name_eq: "parent_name_query",
           parent_articles_title_eq: 'parents_article_title_query'
         }).result
+
         real_query = remove_quotes_and_backticks(s.to_sql)
 
-        expect(real_query)
-          .to match(%r{LEFT OUTER JOIN articles ON (\('default_scope' = 'default_scope'\) AND )?articles.person_id = people.id})
-        expect(real_query)
-          .to match(%r{LEFT OUTER JOIN articles articles_people ON (\('default_scope' = 'default_scope'\) AND )?articles_people.person_id = parents_people.id})
+        if ::Gem::Version.new(::ActiveRecord::VERSION::STRING) > ::Gem::Version.new(Constants::RAILS_6_0)
+          expect(real_query)
+                  .to match(%r{LEFT OUTER JOIN articles ON (\('default_scope' = 'default_scope'\) AND )?articles.person_id = parents_people.id})
+          expect(real_query)
+                  .to match(%r{LEFT OUTER JOIN articles articles_people ON (\('default_scope' = 'default_scope'\) AND )?articles_people.person_id = people.id})
+        else
+          expect(real_query)
+                  .to match(%r{LEFT OUTER JOIN articles ON (\('default_scope' = 'default_scope'\) AND )?articles.person_id = people.id})
+          expect(real_query)
+                  .to match(%r{LEFT OUTER JOIN articles articles_people ON (\('default_scope' = 'default_scope'\) AND )?articles_people.person_id = parents_people.id})
+        end
```

Comment:
If I recall correctly, this is the test that was failing on Rails 6-0-stable. Are you saying that the new query that ransack builds is correct? My understanding was that the CI failure was a real failure that should be fixed.

Reply:
@deivid-rodriguez @varyonic this does seem to be passing the 6.0 CI, great work.

Reply:
I've added an example of a failing test in https://github.com/activerecord-hackery/ransack/issues/1119#issuecomment-630861662